lms/djangoapps/verify_student/views.py

```diff
@@ -1044,85 +1008,52 @@ def results_callback(request):
+
+    Unlike in-course reverification, this flow requires users to submit
+    *both* face and ID photos.  In contrast, during in-course reverification,
+    students submit only face photos, which are matched against the ID photo
+    the user submitted during initial verification.
+
     """
     @method_decorator(login_required)
     def get(self, request):
         """
-        display this view
-        """
-        context = {
-            "user_full_name": request.user.profile.name,
-            "error": False,
-        }
-
-        return render_to_response("verify_student/photo_reverification.html", context)
+        Render the reverification flow.
 
-    @method_decorator(login_required)
-    def post(self, request):
-        """
-        submits the reverification to SoftwareSecure
+        Most of the work is done client-side by composing the same
+        Backbone views used in the initial verification flow.
         """
-
-        try:
-            attempt = SoftwareSecurePhotoVerification(user=request.user)
-            b64_face_image = request.POST['face_image'].split(",")[1]
-            b64_photo_id_image = request.POST['photo_id_image'].split(",")[1]
-
-            attempt.upload_face_image(b64_face_image.decode('base64'))
-            attempt.upload_photo_id_image(b64_photo_id_image.decode('base64'))
-            attempt.mark_ready()
-
-            # save this attempt
-            attempt.save()
-            # then submit it across
-            attempt.submit()
-            return HttpResponseRedirect(reverse('verify_student_reverification_confirmation'))
-        except Exception:
-            log.exception(
-                "Could not submit verification attempt for user {}".format(request.user.id)
-            )
+        status, _ = SoftwareSecurePhotoVerification.user_status(request.user)
+        if status in ["must_reverify", "expired"]:
             context = {
                 "user_full_name": request.user.profile.name,
-                "error": True,
+                "platform_name": settings.PLATFORM_NAME,
+                "capture_sound": staticfiles_storage.url("audio/camera_capture.wav"),
             }
-            return render_to_response("verify_student/photo_reverification.html", context)
-
-
-@login_required
-def reverification_submission_confirmation(_request):
-    """
-    Shows the user a confirmation page if the submission to SoftwareSecure was successful
-    """
-    return render_to_response("verify_student/reverification_confirmation.html")
-
-
-@login_required
-def reverification_window_expired(_request):
-    """
-    Displays an error page if a student tries to submit a reverification, but the window
-    for that reverification has already expired.
-    """
-    # TODO need someone to review the copy for this template
-    return render_to_response("verify_student/reverification_window_expired.html")
+            return render_to_response("verify_student/reverify.html", context)
```

Comment:
[TemplateView](https://docs.djangoproject.com/en/1.8/ref/class-based-views/base/#templateview) could help clean some of this up, but introduces a new problem since you have to switch templates based on the model lookup. I guess you could look that up once and store it as a `cachedproperty`...

```python
class ReverifyView(TemplateView):
  @method_decorator(login_required)
  def dispatch(...)
    return super(...).dispatch(...)

  def get_template_names(self):
    if self.status in ["must_reverify", "expired"]:
      return ['verify_student/reverify.html']
    return ['verify_student/reverify_not_allowed.html']

  # Not sure if cached_property in Django 1.4
  @cached_property
  def status(self):
    status, _ = SoftwareSecurePhotoVerification.user_status(request.user)
    return status

  def get_context_data(...):
    context = super(ReverifyView, self).get_context_data(...)
    context.update({
      'status': self.status,
      'user_full_name': request.user.profile.name,
      'platform_name': settings.PLATFORM_NAME,
      'capture_sound': staticfiles_storage.url("audio/camera_capture.wav"),
    });
    return context
```